FlashCap.Core/Internal/AVFoundation/AVCaptureSession.cs

```diff
@@ -18,66 +18,173 @@ partial class LibAVFoundation
 {
     public sealed class AVCaptureSession : LibObjC.NSObject
     {
+        
+        private AVCaptureVideoDataOutput? _videoDataOutput;
+        private AVCaptureInput? _videoDataInput;
+        
+
         public AVCaptureSession() : base(IntPtr.Zero, false)
         {
             Init();
         }
 
         private void Init()
         {
-            var sessionClass = LibObjC.SendAndGetHandle(
-                LibObjC.GetClass("AVCaptureSession"),
-                LibObjC.GetSelector(LibObjC.AllocSelector));
+            try
+            {
+                var sessionClass = LibObjC.SendAndGetHandle(
+                    LibObjC.GetClass("AVCaptureSession"),
+                    LibObjC.GetSelector(LibObjC.AllocSelector));
 
-            var sessionObj = LibObjC.SendAndGetHandle(
-                sessionClass,
-                LibObjC.GetSelector("init"));
+                var sessionObj = LibObjC.SendAndGetHandle(
+                    sessionClass,
+                    LibObjC.GetSelector("init"));
+                
+                if (sessionObj == IntPtr.Zero)
+                {
+                    throw new Exception("Failed to create AVCaptureSession");
+                }
 
-            Handle = sessionObj;
+                Handle = sessionObj;
 
-            LibCoreFoundation.CFRetain(this.Handle);
+                LibCoreFoundation.CFRetain(this.Handle);
+                
+            } catch (Exception ex)
+            {
+                Console.WriteLine($"Error initializing AVCaptureSession: {ex.Message}");
```

Comment:
Console output may not be readable in some environments.
Output to `Trace` or, if purely for debugging purposes, to `Debug`, or remove it if you don't need it anymore.

(Same as on this source code other locations)